### PR TITLE
Updated SafeMath from the latest OpenZeppelin code

### LINKED ISSUE #378: Update SafeMath

### PR DESCRIPTION
**Pull request by schemar**
https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/math/SafeMath.sol

Fixes #378

### CODE REVIEW

File: contracts/lib/SafeMath.sol [hunk 1]
```diff
@@ -1,6 +1,6 @@
-pragma solidity ^0.4.23;
+pragma solidity ^0.4.24;
```
Comment (deepesh-kn) [inline contracts/lib/SafeMath.sol]: Other solidity files are using 0.4.23. Should we consider using 0.4.24 in all new files?

File: contracts/lib/SafeMath.sol [hunk 2]
```diff
@@ -15,98 +15,128 @@ pragma solidity ^0.4.23;
...
+            c >= a,
+            "Overflow when adding."
+        );
 
         return c;
     }
 
     /**
-    * @notice Divides two numbers and returns the remainder.
+    * @notice Divides two numbers and returns the remainder (unsigned integer
```
Comment (deepesh-kn) [inline contracts/lib/SafeMath.sol]: indentation is incorrect.

File: contracts/lib/SafeMath.sol (hunk 1 above)
Comment (schemar) [reply contracts/lib/SafeMath.sol]: Hm, I don't know. As solidity is developing relatively quickly including the addition of new features, I would say update whatever you work on? What do you think?

Comment (deepesh-kn) [reply contracts/lib/SafeMath.sol]: We can keep `0.4.23`, and when we plan to upgrade we can create a ticket.

**Pull request merged**